fe/fe-core/src/main/java/org/apache/doris/analysis/TimeV2Literal.java

```diff
@@ -0,0 +1,269 @@
+                s = s.substring(0, s.indexOf("."));
+            }
+            int len = s.length();
+            if (len == 1) {
+                s = "00:00:0" + s;
+            } else if (len == 2) {
+                s = "00:00:" + s;
+            } else if (len == 3) {
+                s = "00:0" + s.charAt(0) + ":" + s.substring(1);
+            } else if (len == 4) {
+                s = "00:" + s.substring(0, 2) + ":" + s.substring(2);
+            } else {
+                s = s.substring(0, len - 4) + ":" + s.substring(len - 4, len - 2) + ":" + s.substring(len - 2);
+            }
+            if (sign) {
+                s = '-' + s;
+            }
+            s = s + tail;
+        }
+        if (s.indexOf(':') == s.lastIndexOf(':')) {
+            s = s + ":00";
+        }
+        String[] parts = s.split(":");
+        if (parts.length != 3) {
+            throw new AnalysisException("Invalid format, must have 3 parts separated by ':'");
+        }
+        int scale = 0;
+        try {
+            hour = Float.parseFloat(parts[0]);
+        } catch (NumberFormatException e) {
+            throw new AnalysisException("Invalid hour format", e);
+        }
+
+        try {
+            minute = Long.parseLong(parts[1]);
+        } catch (NumberFormatException e) {
+            throw new AnalysisException("Invalid minute format", e);
+        }
+
+        String[] secondParts = parts[2].split("\\.");
+        if (secondParts.length > 2) {
+            throw new AnalysisException("Invalid second format");
+        }
+
+        try {
+            second = Long.parseLong(secondParts[0]);
+        } catch (NumberFormatException e) {
+            throw new AnalysisException("Invalid second format", e);
+        }
+
+        if (secondParts.length == 2) {
+            String microStr = secondParts[1];
+            scale = microStr.length();
+
+            if (scale > 6) {
+                microStr = microStr.substring(0, 6);
+            }
+
+            StringBuilder sb = new StringBuilder(microStr);
+            while (sb.length() < 6) {
+                sb.append('0');
+            }
+
+            try {
+                microsecond = Long.parseLong(sb.toString());
+            } catch (NumberFormatException e) {
+                throw new AnalysisException("Invalid microsecond format", e);
+            }
+        } else {
+            microsecond = 0L;
+            scale = 0;
+        }
+
+        this.type = ScalarType.createTimeV2Type(scale);
+        if (checkRange(hour, minute, second, microsecond)) {
+            throw new AnalysisException("time literal [" + s + "] is out of range");
+        }
+    }
+
+    @Override
+    protected String toSqlImpl() {
+        return "\"" + getStringValue() + "\"";
+    }
+
+    @Override
+    protected void toThrift(TExprNode msg) {
+        msg.node_type = TExprNodeType.TIMEV2_LITERAL;
+        msg.timev2_literal = new TTimeV2Literal(getStringValue());
```

Comment:
why use string to do SerDe, why not use four attr to present from hour to microsecond? this will lead to be do some useless parsing job